bulk_indexer.go

```diff
@@ -530,10 +539,14 @@ func indexnth(s []byte, nth int, sep rune) int {
 	})
 }
 
-type errorTooManyRequests struct {
-	res *esapi.Response
+type errorFlushFailed struct {
+	resp        string
+	statusCode  int
+	tooMany     bool
+	clientError bool
+	serverError bool
```

Comment:
nice 👍🏼

Reply:
one other option would be to have a typed error for each case, and then wrapping them into an error returned on the specific branch, but we can always change that later